### PR TITLE
Make information about application submissions consistent across the teams app

### LINKED ISSUE #958: Make information about application submissions consistent across the teams app
We have different views where we show information on applications connected to a certain project:

- the project list overview (`/projects`)
  - this lists all projects, is accessible to all users (also non-signed-in users) and shows numbers for applications in draft mode as well as applications that have been submitted for each project
- the single project view (`/projects/id-project-name`)
  - this lists a single project, is accessible to all users (also non-signed-in users) and shows the *total* of applications connected to the project, regardless of whether they've been submitted or not
- the mentor dashboard (`/mentors/applications`)
  - this is accessible to a project submitter only and it lists all *submitted applications* attached to each project of said submitter

As an organiser, I tend to look at the project list overview to see how many projects have how many applications attached to them. But as a mentor, I might (actually, I probably _would_) simply be interested in my own project, meaning I'd get those numbers from my own project page (`/projects/id-project-name`). Unfortunately, that and the mentor dashboard have “conflicting“ information, in the sense that the number of applications shown in those views is not always the same. 

There are different ways to change this:
- remove the numbers of in-progress applications on the project list overview, and only show the number of submitted applications
- move the draft vs. submitted count to the organizer project overview (`organizers/projects`) if this is a number that is more relevant for organizers than anyone else
- add rows in the table on the single project view to split the sum of applications into applications submitted vs. applications in-progress
- clarify the wording in the table on the single project view to explain that this is a TOTAL and not just submitted applications
- probably other possibilities which I'm not thinking of right now

This is not an urgent or critical issue, but something that we should definitely change in the future, depending on what we want the mentors to know and what they are interested in knowing.

### PR DESCRIPTION
**Pull request by klappradla**
This is supposed to fix #958 by:
- distinguishing between *submitted* and *in-progress*  applications on the project show page 
   screen shot 2018-03-06 at 23 22 33

- updating the wording on the mentor's application index page _(the table-caption)_
    screen shot 2018-03-06 at 23 19 02

Aside that it also includes a small *drive-by-fix™* to display the right headlines for fields on the mentor application page if the application has the project as 2nd choice:
screen shot 2018-03-06 at 22 26 49
☝️ this previously always said (`1st project`)

**Feature Specs**
As I said in the previous dev-sync, I think feature specs with headless Chrome could make a nice addition to the app and help to make sure behavior does not break.
I for this PR added all the setup for headless Chrome and Capybara and implemented some tiny feature specs for the things I changed.

**please note:** ☝️ does of course not yet cover the behavior of the respective pages.
But I did not have more time and didn't want to make this PR even huger. I will add some follow up PRs with refactorings and feature specs - but I don't think I'll have any time until the weekend and did not want to leave this lying around until then 😉 

**Up for discussion:**
I changed the configuration for the state machine column in the `ApplicationDraft` model, since I find it hinders test setups a bit... 🙈 I'm free for discussing this / changing this back though.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -40,6 +45,7 @@ deploy:
       repo: rails-girls-summer-of-code/rgsoc-teams
 addons:
   postgresql: '9.5'
+  chrome: stable
```
Comment (carpodaster) [inline .travis.yml]: Would that mean that we don't need to install it manually in `before_install`?

File: .travis.yml [hunk 2]
```diff
@@ -9,12 +8,18 @@ notifications:
...
 before_install:
+  - sudo apt-get update
+  - sudo apt-get install chromium-chromedriver
   - gem update --system
   - gem --version
 before_script:
   - cp config/database.yml.travis config/database.yml
   - bundle exec rails db:create db:schema:load RAILS_ENV=test
   - npm install -g jshint
+  - "export PATH=$PATH:/usr/lib/chromium-browser/"
```
Comment (carpodaster) [inline .travis.yml]: (These CI hacks are why I hate Capybara so much 🙃 )

File: Gemfile [hunk 3]
```diff
@@ -70,4 +70,9 @@ group :test do
   gem 'rspec-html-matchers'
   gem 'rspec-activemodel-mocks'
   gem 'rubocop', require: false
+
+  gem 'capybara'
+  gem 'capybara-screenshot'
+  gem 'selenium-webdriver'
+
```
Comment (carpodaster) [inline Gemfile]: Empty line + can you sort the gems in that block alphabetically?

File: README.md [hunk 4]
```diff
@@ -42,11 +42,12 @@ Requirements:
 
 * PostgreSQL 9.5 or newer
 * Ruby 2.5.0
+* [Chromedriver](https://sites.google.com/a/chromium.org/chromedriver/)
 
 ### Setup on Ubuntu
 ```bash
 # Install required packages
-$ sudo apt-get install postgresql libpq-dev libcurl3 libcurl3-gnutls libcurl4-openssl-dev postgresql-contrib-9.5
+$ sudo apt-get install postgresql libpq-dev libcurl3 libcurl3-gnutls libcurl4-openssl-dev postgresql-contrib-9.5 chromium-chromedriver
```
Comment (carpodaster) [inline README.md]: Were you able to test this on a Linux machine? I cannot verify this until Sunday since I only got a MBP with me right now.

File: README.md [hunk 5]
```diff
@@ -158,6 +158,20 @@ You can optionally create a test-coverage report in `coverage/*` like so:
 
     COVERAGE=yes bundle exec rake spec
 
+Feature tests run in headless Chrome. You can run them in an actual visible window by tagging the examples with `driver: :chrome`, like so:
```
Comment (carpodaster) [inline README.md]: 💟  This is great!

Can you toss in a "_for local debugging puposes_" so that we don't end up with committed `binding.pry`s? ;)

File: app/views/mentors/applications/show.slim [hunk 6]
```diff
@@ -8,14 +8,22 @@ h2
   small For #{link_to @application.project_name, project_path(@application.project_id)} #{icon('star') if @application.first_choice?}
 
 div.panel.panel-default
-  div.panel-heading #{ApplicationDraft.human_attribute_name :why_selected_project1}
+  div.panel-heading
+    - if @application.first_choice?
```
Comment (carpodaster) [inline app/views/mentors/applications/show.slim]: Can we make this a helper method?

File: app/views/mentors/applications/show.slim [hunk 7]
```diff
@@ -8,14 +8,22 @@ h2
...
+    - else
+      = ApplicationDraft.human_attribute_name :why_selected_project2
   div.panel-body
-    p #{@application.why_selected_project}
+    p = @application.why_selected_project
 
 div.panel.panel-default
-  div.panel-heading #{ApplicationDraft.human_attribute_name :plan_project1}
+  div.panel-heading
+    - if @application.first_choice?
```
Comment (carpodaster) [inline app/views/mentors/applications/show.slim]: See above: move to helper method?

File: app/views/projects/show.slim [hunk 8]
```diff
@@ -15,41 +13,7 @@ nav.page
...
-  tr
-    th Applications (2nd Choice)
-    td = @project.second_choice_application_drafts.count
-  tr
-    th Code of Conduct
-    td = link_to @project.code_of_conduct, @project.code_of_conduct
-  tr
-    th License
-    td = @project.license
+= render 'project_info', project: @project
```
Comment (carpodaster) [inline app/views/projects/show.slim]: or shall we name the partial `_project.html.slim` and define it as the canonical way to display a project's details with its `to_partial_path`:

```
= render @project
```

File: spec/features/mentors/applications/index_spec.rb [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+RSpec.describe 'Application index page', type: :feature do
+  let(:user)         { create(:user) }
+  let(:project)      { create(:project, :in_current_season, :accepted, submitter: user) }
+  let(:team1)        { create(:team, name: 'We could be Heroines') }
+  let(:team2)        { create(:team, name: 'Electric Queen') }
+  let(:application1) { create(:application, :in_current_season, :for_project, project1: project, team: team1) }
+  let(:application2) { create(:application, :in_current_season, :for_project, project2: project, team: team2) }
+
+  # TODO: use timecop to make sure the site is available during season
```
Comment (carpodaster) [inline spec/features/mentors/applications/index_spec.rb]: Is this a TODO for this PR?

File: spec/features/mentors/applications/index_spec.rb [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+  context 'when mentoring a single projcet' do
+    before do
+      application1
```
Comment (carpodaster) [inline spec/features/mentors/applications/index_spec.rb]: Turn these into `let!`

**Comment by F3PiX** [comment]
As we run Rails 5 and Rails 5 ships with Capybara, AND claims to manage everything for us under the hood, I am a bit surprised that we need all the tweaks.
 Did you consider just generating the system specs for RSpec?  (I think, IIRC, that that would install Launchy and also takes care of the screenshots and the headless browser).

???

**Comment by carpodaster** [comment]
@f3pix I haven‘t used Rails 5.1’s system tests, yet. But I thought capybara is not shipped with it and you‘re on your own making it work on your machine. That was May biggest beef with system tests‘ claim that it would Just Work(tm). Also, a recent Rails 5 greenfield app was generated with a `config.generate_system_tests = false` (or something like this) in application.rb

**Comment by F3PiX** [comment]
Yeah, Just Work == It's not a bug, it's a feature.
I am using Rails/RSpec system tests in my latest project, and [5.1 is shipped with Capybara](http://edgeguides.rubyonrails.org/5_1_release_notes.html#system-tests). 

Then again, I am not using CI, and that needs [some configuration, according to the guide I think I used. ](https://medium.com/table-xi/a-quick-guide-to-rails-system-tests-in-rspec-b6e9e8a8b5f6) (I can send you a link to my repo, if it helps to see a bit more than a sandbox app. )

Seeing the discussion, I wouldn't mind if we open a separate PR for the introduction of the feature/system tests. And not mix it with the application stuff.
And then maybe we introduce it conservatively - first in dev env.
Are you open to that, @klappradla ? @carpodaster ?

File: README.md (hunk 5 above)
Comment (F3PiX) [reply README.md]: Cool! I didn't know that 💃

File: app/views/mentors/applications/index.slim [hunk 11]
```diff
@@ -1,10 +1,10 @@
-ul[class=:breadcrumb]
+ul.breadcrumb
   li = link_to 'Applications', [:mentors, :applications]
-  
-p Get insight in certain fields of teams applying to work on your project this summer. You can take notes for yourself and pick your favorite applications; the sign off tells us that you believe the team has the needed requirements and skills to work on the project, and faving will make the application more likely to be accepted and having the team work on your project this summer.
+
+p = t('.explanation')
```
Comment (F3PiX) [inline app/views/mentors/applications/index.slim]: ❤️

File: app/views/projects/_project_info.slim [hunk 12]
```diff
@@ -0,0 +1,44 @@
+ruby:
+  project = local_assigns.fetch(:project)
+
+table.table.table-striped.table-bordered.qa-project-info
+  tr
+    th Mentor
+    td
+      = link_to project.mentor_name, project.mentor rescue project.mentor_name
```
Comment (F3PiX) [inline app/views/projects/_project_info.slim]: I am not familiar with `rescue` in the view. Is it rescue-ing the link? 🤔 
(And if so, wouldn't it be needed in the next link on line 10 as well?

File: spec/support/capybara.rb [hunk 13]
```diff
@@ -0,0 +1,28 @@
+require 'capybara-screenshot/rspec'
+require 'selenium/webdriver'
+
+Capybara.javascript_driver = :headless_chrome
+
+Capybara.configure do |config|
+  config.default_driver = :headless_chrome
+  config.javascript_driver = :headless_chrome
+
```
Comment (F3PiX) [inline spec/support/capybara.rb]: Do we need both the js_driver configs, in Line 4 _and_ line 8?

**Comment by F3PiX** [comment]
Great improvements! 🙇

File: .travis.yml (hunk 1 above)
Comment (klappradla) [reply .travis.yml]: As I said, I threw this together pretty quickly just to show how it could eventually look like 🙈 Sorry for the messy state this is in.
For the installation, I followed the [Travis Docs](https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-the-Chrome-addon-in-the-headless-mode) (basically coming from [this Thoughtbot blogpost](https://robots.thoughtbot.com/headless-feature-specs-with-chrome) and the setups I use at work).

As of the docs, both would be necessary, but I think one should definitely do some more experiments there, since it looks suspiciously complex 👀

File: .travis.yml (hunk 2 above)
Comment (klappradla) [reply .travis.yml]: 😹 we could probably even ditch this, since the app is not very JS heavy. It's just a default we always have at work, so I thought why not.

File: README.md (hunk 4 above)
Comment (klappradla) [reply README.md]: No, did not try this out on a Linux machine, just on a trusty Ubuntu Docker container some weeks ago.

File: spec/features/mentors/applications/index_spec.rb (hunk 9 above)
Comment (klappradla) [reply spec/features/mentors/applications/index_spec.rb]: Depends 😉 

I thought if we agree on feature specs being useful, I'd update the ones for the mentor bits of the app to actually test more than just the tiny changes from this PR and then also address the timing things there.

File: spec/features/mentors/applications/index_spec.rb (hunk 10 above)
Comment (klappradla) [reply spec/features/mentors/applications/index_spec.rb]: 💡 having solely worked on projects where Rubocop discourages the use of `let!` I actually forgot about it 🙈

File: app/views/projects/_project_info.slim (hunk 12 above)
Comment (klappradla) [reply app/views/projects/_project_info.slim]: Hm, interesting question! I tbh have no idea what this is doing even though I may have written it myself some time ago...

File: spec/support/capybara.rb (hunk 13 above)
Comment (klappradla) [reply spec/support/capybara.rb]: Probably not - thanx for pointing out 🙏

**Comment by klappradla** [comment]
First of all: sorry for this super messy PR @carpodaster & @F3PiX 🙈 

I threw it together pretty quickly and should definitely clean up a bit more until this is ready to be merged 🌵
The `driver: :chrome` thing I added to the readme can come in really handy in my opinion - it's basically how I work when I have to do non-JS things that touch Rails views.

**system tests**
For what you wrote on system tests @F3PiX : thanx for that 🙏 
I didn't know RSpec makes a difference between them and always just assumed, `feature` would wrap what Rails calls *"system"*. 
Not sure if I really know their application yet though 🙈 . Should the tests I added in this PR be converted into `system` ones?

☝️ tl;dr
> I didn't know that there's a difference between `feature` and `system` test and now that I know don't know the actual difference.

I also did not know about Rails shipping with Capybara and a collection of headless browsers - that's awesome 🎉 Does this mean I can ditch anything I did for setting up headless Chrome?

**setup**
For what you wrote on the file extensions @carpodaster, yes it's true and as far as I know does not do any damage to append a `.html` to the filenames for `.slim` files. It's visual clutter to me, therefore I removed it - but happy to but it back ✌️ I totally wasn't aware that people may actually care about it 🙈 

For headless Chrome setup: you're definitely right, it will add another burden to get set up and running with the app 😞 . I thought about it and considered it an okish trade off compared to the benefit in terms of productivity and maintainability. It doesn't hinder people from running the app locally and getting started with the code at first... so they may be motivated enough to dive into these details once they're on the verge of opening a PR 🤔 

I think™ one needs to a) have Chrome installed b) have the `chromedriver` installed (homebrew as you said) and c) start the chromedriver.
If I look at my processes:
```
❯ ps aux | ag chrome
max              57906   0.0  0.1  4446784  18320 s007  S     3:39PM   0:00.71 /usr/local/bin/chromedriver --port=9519
max              56795   0.0  0.1  4418684  14752 s007  S     3:35PM   0:00.16 /usr/local/bin/chromedriver --port=9518
max              55833   0.0  0.1  4418684  11840 s007  S     3:31PM   0:00.14 /usr/local/bin/chromedriver --port=9517
max              54972   0.0  0.1  4418684  11864 s007  S     3:26PM   0:00.13 /usr/local/bin/chromedriver --port=9516
max              54357   0.0  0.1  4418684  11864 s007  S     3:23PM   0:00.14 /usr/local/bin/chromedriver --port=9515
max              57907   0.0  0.0        0      0 s007  Z     3:39PM   0:00.00 (Google Chrome)
max              56796   0.0  0.0        0      0 s007  Z     3:35PM   0:00.00 (Google Chrome)
max              55834   0.0  0.0        0      0 s007  Z     3:31PM   0:00.00 (Google Chrome)
max              54973   0.0  0.0        0      0 s007  Z     3:26PM   0:00.00 (Google Chrome)
max              12881   0.0  0.0  4259132    296 s018  R+   11:26PM   0:00.00 ag chrome
max              54358   0.0  0.0        0      0 s007  Z     3:23PM   0:00.00 (Google Chrome)
```

You see `chromedriver` at multiple ports.

I use the `plist` files that come with homebrew to start it once:
```
brew services start chromedriver
```

it for me is then automatically started and stopped whenever I run tests:
```
❯ brew services list
Name         Status  User Plist
postgresql   started max  /Users/max/Library/LaunchAgents/homebrew.mxcl.postgresql.plist
redis        started max  /Users/max/Library/LaunchAgents/homebrew.mxcl.redis.plist
emacs        stopped
memcached    started max  /Users/max/Library/LaunchAgents/homebrew.mxcl.memcached.plist
chromedriver stopped
```

➕ I agree I should look for an easier setup for the Chrome & the feature tests.
I also came across [chromedriver as a gem](https://github.com/flavorjones/chromedriver-helper) but considered it to be the wrong tool for the job (even worse then packaging JS  CSS in gems).

**so...**
I'll 👀 for an easier setup of the whole thing plus clean up this PR. Happy about suggestions 💡 (system test stuff and other headless browsers)

**Comment by F3PiX** [comment]
The difference, as I understand it, is that 'feature tests' come from Capybara itself, and `system tests` are feature tests integrated in the Rails system. (IIRC systems tests work with AR, and that would be the reason why the database cleaner is not needed for system tests.) 

I checked my setup (in a new Rails 5.1 app out of the box).
The gems that came with it are capybara and selenium-webdriver. (And  RSpec 3.7.0+, which introduced the system specs. We have that in the Teams app too. )
http://rspec.info/blog/2017/10/rspec-3-7-has-been-released/

This is the only config I needed for RSpec in `rails_helper.rb` 
```
+  # Using system tests in Rspec
+  config.before(:each, type: :system) do
+    driven_by :rack_test
+  end
+
+  config.before(:each, type: :system, js: true) do
+    driven_by :selenium_chrome_headless
+  end
 ```
Additionally, we'll need a `/system` dir in `/rspec`

And yes,  just replace type :feature with :system in the specs. The system specs accept both the `scenario` syntax and the `context/it`. I think I even mix them. 

The selenium_chrome_headless driver probably requires Chromium; I am not sure if `brew chromedriver` resolves that.  (My guess would be: not).
The :rack_test is the default test driver, and said to be fast. You just don't see the browser open itself and walk through the pages. 

As we don't rely on JS that much, installation may be much easier when we skip the JS tests for now, so we don't need `selenium_chrome_headless`, and give us some slack to get used to system tests. 

But but but what about the screenshots? Well, you can put `take_screenshot` within a test. And even better, it always creates a screenshot when tests are failing. 😍     

Note: I didn't use JS system tests yet, but mainly because I accidentally decided to use React 🙈  and  I am pretty sure that doesn't work with system tests.

Food for thought.

**Comment by klappradla** [comment]
Ok, good to know @F3PiX 🙏 

Sidenote: React and feature / system tests (at least with chromedriver) does of course work - and you should probably use them 😉 

Does the `selenium_chrome_headless` also allow you to open up a "real", visible Chrome window and follow / debug there? I personally find it super hard, if not impossible if I cannot use the browser and its dev-tools to write feature specs. Relying on screenshots or the plain html output oft the page feels a bit like running around in complete darkness ⚫️

File: app/views/projects/_project_info.slim (hunk 12 above)
Comment (carpodaster) [reply app/views/projects/_project_info.slim]: My guess is that `link_to` buckles when `project.mentor` is `nil`. In that case it only displays the name w/o a link?

**Comment by klappradla** [comment]
_disclaimer: I'm def. not at my mental heights today, so I may be missing something obvious..._

So @F3PiX & @carpodaster, I revisited the feature / system test setup, read some things about it and tried to understand it and in consequence simplify what I initially came up with.

**feature vs. system**
I went with @F3PiX's suggestion to move things over to `type: :system` tests. The only difference this had to `feature` ones was that Capybara would default to a driver that can handle JavaScript...
I in the end turned them back into `feature` because I prefer the term over `system`. Can also change it back again, no super strong feelings here.

**chromedriver**
As it turns out, Selenium itself actually requires `chromedriver` to be installed. Check https://github.com/SeleniumHQ/selenium and for more details (you can also browse the issues for it) the [Capybara Repo](https://github.com/teamcapybara/capybara#selecting-the-driver).

Setting the driver to `selenium` requires a weird headless sort of Firefox - but I did not even try to install it, since it's less well documented than getting `chromedriver` set up and I did not see how this could be easier for newcomers.

**no js**
I also did some experiments using the `:rack_test` driver which apparently does not support any JavaScipt. But it's super hard to write any tests that pass for this driver - and there's no possibility for debugging with an actual window... 😞 I could not get any of my tests to pass with this driver.
But tbh. I also did not max. out trying this, since I again don't really see the point in using a driver that is so far a away from the actual real-world-usage of the app.

So what do you think about the setup now? Does this make more sense for you?

PS: @carpodaster I did not implement all your suggestions yet, wanted to investigate into the testing stuff first.

File: app/views/mentors/applications/show.slim (hunk 6 above)
Comment (klappradla) [reply app/views/mentors/applications/show.slim]: Hm, I think I don't really get what you have in mind 🙈 

I'll just interpolate the key - that also removes the conditional and drys up the view. Fine?

File: spec/features/mentors/applications/index_spec.rb (hunk 9 above)
Comment (klappradla) [reply spec/features/mentors/applications/index_spec.rb]: Well, I added the time travel now. It's not really necessary for the specs to work - when I added the comment, I just thought it would make the specs more close to _"the real world™"_, since these bits of the app are only used during a certain phase of the season...

Do you think this makes sense? Or should I get rid of the time travel again to make the tests shorter / simpler?

**Comment by klappradla** [comment]
Ok, pushed some more commits with the changes you requested @carpodaster ✌️ 

Also did some quick investigation on the `Ubuntu` setup, this you mentioned this earlier 🤓 

### tl;dr
it works like charm 🖤 
(but I'm def. no expert with Ubuntu 😂 )

### what I did...

So I set up a blank Ubuntu Docker container: _(dockerfile is a rip off from somewhere on Github)_
```dockerfile
FROM ubuntu:17.10

RUN \
  apt-get update && \
  apt-get -y upgrade && \
  apt-get install -y build-essential && \
  apt-get install -y curl git unzip vim wget

ENV HOME /root
WORKDIR /root

# Define default command.
CMD ["bash"]
```

build it:

    docker build -t tryout .

then ran it:

    docker run -it tryout

then cloned the repo, installed all dependencies (with a slight change for postgres and adding `node-js`)

    apt-get install postgresql libpq-dev libcurl3 libcurl3-gnutls libcurl4-openssl-dev postgresql-contrib chromium-chromedriver node-js

⚠️  then could not get `ruby 2.5` on Ubuntu and installed `rbenv` and `ruby-build` from github directly... 🙈 

set up the db related stuff:

    /etc/init.d/postgresql start

    su postgres
    createuser max -s -P
    exit

set up the app
```bash
cd rgsoc
gem install bundler
cp config/database.yml.example config/database.yml # and adapt
bundle
```

Then set up the db and ran the specs:
```
root@1daa068ae8ab:~/rgsoc-teams# bundle exec rails db:create
Created database 'rgsocteams_development'
Created database 'rgsocteams_test'
root@1daa068ae8ab:~/rgsoc-teams# bundle exec rspec

Randomized with seed 9506
......................................................................................................................................................................
......................................................................................................................................................................
......................................................................................................................................................................
......................................................................................................................................................................
......................................................................................................................................................................
......................................................................................................................................................................
......................................................................................................................................................................
......................................................................................................................................................................
............................................................

Finished in 1 minute 32.18 seconds (files took 4.31 seconds to load)
1388 examples, 0 failures

Randomized with seed 9506

root@1daa068ae8ab:~/rgsoc-teams
```

🎉 So installing `chromedriver` is enough - we don't even need the actual Chrome app.

**Comment by klappradla** [comment]
Ping @F3PiX & @carpodaster 👋 

How are your feelings on merging / not-merging this? Anything else you want me to try out or do differently?

**Comment by F3PiX** [comment]
Sorry @klappradla , I meant to look into this, but my head was overloaded with other code :-)
I am really surprised about the issues you ran into with the set up.  I'd like to dig a bit deeper into the 'open Chrome window', because it seems unlikely that it is not working with Rails's Capybara. Right? 

I take it you are not into my earlier suggestion: 

> Seeing the discussion, I wouldn't mind if we open a separate PR for the introduction of the feature/system tests. And not mix it with the application stuff. 

?

I really like to test in my own app if I can get the 'open Chrome window' working. I can do that  this week/somewhere in the next days. And because my app started as a 5+ app, it could be a benchmark for the set up. Which would make it easier for me to follow and reason about it. 

But I am not aware if the Application stuff in this PR is urgent, would it harm to wait another day or 2? Would be easier than split the PR?

**Comment by klappradla** [comment]
No worries @F3PiX 💚 

I guess the main place this PR targeted was the mentors' view on applications. Since this phase is basically over, there's no pressure to merge the functionality and tests in two different PRs I think...

For what you wrote above: I'm not sure if I get it right 🙈 

Which issue did I run into with Chrome windows?
I just changed the setup to use the build in selenium drivers instead of a custom configured headless Chrome and Chrome. I also updated the `README.md` for what to do in order to change the driver from _"headless"_ to _"an actual window"_ - Selenium ships with configuration for both, the headless Chrome as well as the "normal" one.

File: spec/features/mentors/applications/index_spec.rb (hunk 9 above)
Comment (carpodaster) [reply spec/features/mentors/applications/index_spec.rb]: I don't mind either. If it's not required, might as well remove it. But then again, you already added it and it simulates our use case closer. `/shrug`

File: app/views/mentors/applications/show.slim (hunk 6 above)
Comment (carpodaster) [reply app/views/mentors/applications/show.slim]: tbh I don't know what I meant with this, either :see_no_evil:   Ignore it :)

**Comment by klappradla** [comment]
Ok, I rebased this again to current `master`. So if there aren't any different suggestions for the testing story, let's roll with this for now and act accordingly if this turns out to be super painful to set up for students ✌️
